v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -476,53 +508,73 @@ func (r *AzureDeploymentReconciler) handlePollerFailed(ctx context.Context, err
+		result.RequeueAfter = randextensions.Jitter(r.rand, *r.config.SyncPeriod, 0.1)
+	}
+	return result
+}
+
+func (r *AzureDeploymentReconciler) handleSkipReconcile(ctx context.Context) (ctrl.Result, error) {
+	reconcilePolicy := r.GetReconcilePolicy()
 	r.log.V(Status).Info(
-		"Resource successfully created",
+		"Skipping creation of resource due to policy",
+		ReconcilePolicyAnnotation, reconcilePolicy,
 		"resourceID", genruntime.GetResourceIDOrDefault(r.obj))
 
-	ready := r.PositiveConditions.Ready.Succeeded(r.obj.GetGeneration())
-	resourceID, hasResourceID := genruntime.GetResourceID(r.obj)
-	if !hasResourceID {
-		return ctrl.Result{}, errors.Errorf("poller succeeded but resource has no resource id")
+	err := r.updateStatus(ctx)
+	if err != nil {
+		if genericarmclient.IsNotFoundError(err) {
+			conditions.SetCondition(
+				r.obj,
+				r.PositiveConditions.Ready.AzureResourceNotFound(r.obj.GetGeneration(), err.Error()))
+			r.ClearPollerResumeToken()
+			err = r.CommitUpdate(ctx)
+			if err != nil {
+				// NotFound is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
+				// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
+				return ctrl.Result{}, client.IgnoreNotFound(err)
+			}
+		}
+		return ctrl.Result{}, err
 	}
 
-	// TODO: I think if we're really we can avoid this as the HTTP payload in poller should have all we need already?
-	// TODO: for now doing this though because my quick read of the Go SDK is that it does a final GET on the URL too...
-	// TODO: Maybe some Azure services or types of async operations actually don't return the latest resource?
-	status, _, err := r.getStatus(ctx, resourceID)
+	r.ClearPollerResumeToken()
+	conditions.SetCondition(r.obj, r.PositiveConditions.Ready.Succeeded(r.obj.GetGeneration()))
+	err = r.CommitUpdate(ctx)
 	if err != nil {
-		return ctrl.Result{}, errors.Errorf("error getting status for resource ID %q", resourceID)
+		// NotFound is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
+		// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	// Modifications that impact status have to happen after this because this performs a full
-	// replace of status
-	if status != nil {
-		// SetStatus() takes care of any required conversion to the right version
-		err = r.obj.SetStatus(status)
-		if err != nil {
-			return ctrl.Result{},
-				errors.Wrapf(err, "setting status on %s", r.obj.GetObjectKind().GroupVersionKind())
-		}
+	return r.makeSuccessResult(), nil
+}
+
+func (r *AzureDeploymentReconciler) handlePollerSuccess(ctx context.Context) (ctrl.Result, error) {
+	r.log.V(Status).Info(
+		"Resource successfully created",
+		"resourceID", genruntime.GetResourceIDOrDefault(r.obj))
+
+	err := r.updateStatus(ctx)
+	if err != nil {
```

Comment:
same here, since we're not using the lhs anywhere but inside the if block, below syntax would avoid the extra variable assignment lines.
`if err = r.updateStatus(ctx); err != nil {`